tests/sentry/workflow_engine/handlers/detector/test_metric_issue_detector_handler.py

```diff
@@ -1,23 +1,44 @@
-from datetime import timedelta
+from datetime import UTC, datetime, timedelta
+from unittest.mock import patch
 
-from sentry.incidents.grouptype import MetricIssueDetectorHandler
+from sentry import eventstore
+from sentry.eventstream.types import EventStreamEventType
+from sentry.incidents.grouptype import MetricIssue, MetricIssueDetectorHandler
 from sentry.incidents.utils.constants import INCIDENTS_SNUBA_SUBSCRIPTION_TYPE
+from sentry.incidents.utils.types import QuerySubscriptionUpdate
+from sentry.issues.issue_occurrence import IssueOccurrence
+from sentry.issues.producer import PayloadType, produce_occurrence_to_kafka
 from sentry.snuba.dataset import Dataset
 from sentry.snuba.models import SnubaQuery, SnubaQueryEventType
 from sentry.snuba.subscriptions import create_snuba_query, create_snuba_subscription
+from sentry.tasks.post_process import post_process_group
 from sentry.testutils.helpers.datetime import freeze_time
+from sentry.testutils.helpers.features import with_feature
+from sentry.workflow_engine.models import DataCondition, DataPacket
 from sentry.workflow_engine.models.data_condition import Condition
-from sentry.workflow_engine.types import DetectorPriorityLevel
-from tests.sentry.workflow_engine.handlers.detector.test_base import BaseDetectorHandlerTest
+from sentry.workflow_engine.types import (
+    DetectorEvaluationResult,
+    DetectorGroupKey,
+    DetectorPriorityLevel,
+)
+from tests.sentry.workflow_engine.test_base import BaseWorkflowTest
 
 
 @freeze_time()
-class TestEvaluateMetricDetector(BaseDetectorHandlerTest):
+class TestEvaluateMetricDetector(BaseWorkflowTest):
```

Comment:
In order to be able to use the `MetricIssue` type I had to avoid using `BaseDetectorHandlerTest` since it mocks the issue registry

Reply:
🤔 i think this might be a smell that we shouldn't add these tests here.

instead, could we add them to a test in the `incidents` module? then we can just have a new test case and the other detector tests can remain more generic using the mock.